### PR TITLE
Port to serverless: Add Osquery tab to host monitoring page

### LINKED ISSUE #3954: Port Osquery host details to serverless docs
### Description

Stateful docs have content that is not available in serverless docs:

https://www.elastic.co/guide/en/observability/current/analyze-hosts.html#view-host-details

image

https://docs.elastic.co/serverless/observability/analyze-hosts#view-host-details

image

### Resources

n/a

### Which documentation set does this change impact?

Serverless only

### Feature differences

n/a

### What release is this request related to?

N/A

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** @roshan-elastic 

**Stakeholders:** n/a

### PR DESCRIPTION
**Pull request by dedemorton**
## Description
Adds Osquery content to serverless. Note that this is just a port of existing stateful content. It has not been validated because the assumption is that the stateful docs are correct.

If they are not correct, we will need to update stateful and serverless at a later date.

Preview link (you'll need to Expand the Osquery tab to see the docs):
https://elastic-dot-co-docs-production-26non2x4r-elastic-dev.vercel.app/current/serverless/observability/analyze-hosts#view-host-details 

### Documentation sets edited in this PR

_Check all that apply._

- [ ] Stateful (`docs/en/observability/*`)
- [x] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue
Closes #3954

## Checklist

- [ ] Product/Engineering Review
- [ ] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [x] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [ ] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-lwfdr5dvh-elastic-dev.vercel.app

* Built with commit c8841276d45ec42759bb2d66d1fd768bdccfe223

> Issues? Visit #next-docs in Slack

File: docs/en/serverless/transclusion/host-details.mdx [hunk 1]
```diff
@@ -99,5 +101,36 @@ You can also select **Actions** → **Show in Inventory** to view the host Inven
 
 </DocAccordion>
 
-{/* TODO: Find out if OSQuery tab will be included in serverless. It does not currently appear in serverless builds */}
+<DocAccordion buttonContent="Osquery">
+
+{/* Roles role="Admin" goal="use this feature"  */}
```
Comment (dedemorton) [inline docs/en/serverless/transclusion/host-details.mdx:106]: @roshan-elastic For serverless, do you know what role is required to use Osquery? Is it Admin? Editor?

Comment (roshan-elastic) [reply docs/en/serverless/transclusion/host-details.mdx:106]: @chrisdistasio @smith - You don't happen to know do you?

**Comment by dedemorton** [comment]
Update: Waiting to hear back from Tomasz Ciecierski about the serverless role required to use Osquery.

**Comment by dedemorton** [comment]
Heard back from @tomsonpl  about required privileges. It sounds like the requirements are kind of murky (because the privileges underlying the roles are a bit nebulous). Here's what he suggests documenting:

**Admin Role:** Full access to project configuration, including the ability to install, manage, and run Osquery queries through Elastic Agent. This role supports both ad hoc (live) queries and scheduled queries against monitored hosts. Admins can view and analyze the results directly in Elasticsearch.
**Editor Role:** Limited access - they can run queries if pre-configured, may have restricted permissions for setting up and scheduling new queries, especially those that require broader access or permissions adjustments.
**Viewer Role:** Read-only access to data, including viewing Osquery results if configured by users with higher permissions, but they cannot initiate or schedule Osquery queries themselves.

We can fine tune this later if someone can tell us which privileges are help by the Observability rules in serverless.

**Comment by dedemorton** [comment]
@smith @tomsonpl  I've updated this PR with the roles required to use Osquery in serverless.

Can we get someone who knows serverless to review this PR and validate that it's correct? I don't have an environment set up where I can test this.

**Review by smith — APPROVED** [review]
Looks good. Reviewed in Serverless QA and things seem to match up.

**Pull request merged**